server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNodes.java

```diff
@@ -185,11 +186,15 @@ public Map<String, DiscoveryNode> getCoordinatingOnlyNodes() {
         return filteredNodes(nodes, n -> n.canContainData() == false && n.isMasterNode() == false && n.isIngestNode() == false);
     }
 
+    // Ugly hack: when https://github.com/elastic/elasticsearch/issues/94946 is fixed, remove the sorting by ephemeral ID here
+    private final Comparator<DiscoveryNode> MASTERS_FIRST_COMPARATOR = Comparator.<DiscoveryNode>comparingInt(n -> n.isMasterNode() ? 0 : 1)
```

Comment:
I believe `static` is missed here since the comparator is stateless

Reply:
Ah oops yes I meant `static`